laba3_1/main.cpp

```diff
@@ -0,0 +1,138 @@
+#include <iostream>
+#include <ctime>
+#include <vector>
+int GetElementOfMatrix(int **&matrix, int i, int j) {
+  if ((i + j)%2==0) {
+	return 10;
+  } else
+
+	return matrix[i][j/2];
+
+}
+//int SwapColumnsOfMatrics(int **&matrics,int i,int j){
+//
+//}
+
+void EnterRandomMatrix(int **&matrix,int I,int J){
+	int min_r,max_r;
+	srand(time(nullptr));
+	std::cout<<"Enter min element for random"<<std::endl;
+	std::cin>>min_r;
+	std::cout<<"Enter max element for random"<<std::endl;
+	std::cin>>max_r;
+	int m;
+    if(J%2!=0)
+    {
+        m=(J/2+1);
+    }
+    else{
+       m=J/2;
+    }
+    for(int i=0; i<I; i++){
+      for(int j=0; j<m;j++ ){
+
+        matrix[i][j]=min_r+rand()%(max_r+1-min_r);
+
+      }
+    }
+    return;
+}
+void NumberOfRow(int**&matrics, int I, int J){
```

Comment:
Would be easier to use `GetElementOfMatrix` function and work with matrix as it full stored in memory